releasenotes-builder/src/main/java/com/github/checkstyle/MainProcess.java

```diff
@@ -79,7 +80,82 @@ public static List<String> runPostGenerationAndPublication(
+        }
+        errors.addAll(runPostPublication(cliOptions));
+        return errors;
+    }
+
+    /**
+     * Validate that pom version matches issues labels in release notes.
+     *
+     * @param releaseNotes map of release notes messages.
+     * @param cliOptions command line options.
+     * @return list of notes validation errors.
+     */
+    private static List<String> validateNotes(Multimap<String, ReleaseNotesMessage> releaseNotes,
+                                              CliOptions cliOptions) {
+        final String releaseVersion = cliOptions.getReleaseNumber();
+        final long amountOfCommas = getAmountOfCommasInReleaseVersion(releaseVersion);
+
+        final boolean containsNewOrBreakingCompatabilityLabel =
+            releaseNotes.containsKey(Constants.NEW_FEATURE_LABEL)
+                || releaseNotes.containsKey(Constants.NEW_MODULE_LABEL)
+                || releaseNotes.containsKey(Constants.BREAKING_COMPATIBILITY_LABEL);
+
+        final List<String> errors = new ArrayList<>();
+
+        if (isMinor(amountOfCommas) && containsNewOrBreakingCompatabilityLabel) {
+            errors.add(
+                String.format("Validation of release number failed. Release number is "
+                    + "minor(%s), but release notes contain 'new' or "
+                    + "'breaking compatability' labels.", releaseVersion)
+            );
+        }
+        else if (isPatch(amountOfCommas) && !containsNewOrBreakingCompatabilityLabel) {
+            errors.add(
+                String.format("Validation of release number failed. Release number is a "
+                    + "patch(%s), but release notes do not contain 'new' or 'breaking "
+                    + "compatability' labels.", releaseVersion)
+            );
+        }
+
+        return errors;
+    }
+
+    /**
+     * Retrieves the amount of commas in the release version. For instance,
+     * 10.3.3 would return {@code 2}.
+     *
+     * @param releaseVersion release number in the command line interface.
+     * @return amount of commas in release version.
+     */
+    private static long getAmountOfCommasInReleaseVersion(String releaseVersion) {
+        return releaseVersion.chars().filter(character -> character == '.').count();
+    }
+
+    /**
+     * Returns {@code true} if amount of commas is 2. This means
+     * the release is minor.
+     *
+     * @param amountOfCommas amount of commas in release number.
+     * @return {@code true} if amount of commas is 2.
+     */
+    private static boolean isMinor(long amountOfCommas) {
+        return amountOfCommas == 2;
+    }
+
+    /**
+     * Returns {@code true} if amount of commas is 1. This means
+     * the release is a patch.
+     *
+     * @param amountOfCommas amount of commas in release number.
+     * @return {@code true} if amount of commas is 1.
+     */
+    private static boolean isPatch(long amountOfCommas) {
+        return amountOfCommas == 1;
```

Comment:
One comma is minor.
https://semver.org/
https://github.com/checkstyle/checkstyle/issues/3709

Reply:
Didn't know that. Thanks. Should be fixed. Done.